### ISSUE TITLE
Cookie header is not created correctly for AWS Lambda

### ISSUE THREAD

**Issue by dkpun** [remix-run/remix#9513]
### Reproduction

- Create a remix app that saves multiple cookies
- Deploy the app to AWS lambda
- Try to read back the cookies in the app

### System Info

```shell
System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 95.11 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.1 - ~/.volta/tools/image/node/20.11.1/bin/node
    Yarn: 1.22.22 - ~/.volta/tools/image/yarn/1.22.22/bin/yarn
    npm: 9.8.1 - ~/myapp/node_modules/.bin/npm
  Browsers:
    Chrome: 125.0.6422.113
    Safari: 17.4.1
  npmPackages:
    @remix-run/architect: 2.9.2 => 2.9.2
    @remix-run/css-bundle: 2.9.2 => 2.9.2
    @remix-run/dev: 2.9.2 => 2.9.2
    @remix-run/eslint-config: 2.9.2 => 2.9.2
    @remix-run/node: 2.9.2 => 2.9.2
    @remix-run/react: 2.9.2 => 2.9.2
    @remix-run/serve: 2.9.2 => 2.9.2
    @remix-run/web-fetch: 4.4.2 => 4.4.2
```

### Used Package Manager

yarn

### Expected Behavior

You should get back all the saved cookies.

### Actual Behavior

All attempts to read the cookies return null

Looks like this issue was [fixed](https://github.com/remix-run/remix/issues/1663) before but has resurfaced

**Comment by miniplus** [comment]
Not sure on the cause but it seems I'm having the same issue.

Writing and reading locally seems to be just fine.
Writing seems fine on AWS Lambda, Reading does not.

**Comment by dkpun** [comment]
Fixed in v2.10.3
